src/controllers/handlers/entities-handler.ts

```diff
@@ -0,0 +1,118 @@
+import { isErrorWithMessage } from '@dcl/core-commons'
+import { InvalidRequestError } from '@dcl/platform-server-commons'
+import { ForbiddenError } from '../errors'
+import type { FormHandlerContextWithPath } from '../../types'
+
+/**
+ * Handler for the POST /content/entities endpoint
+ *
+ * Validates authorization and proxies entity uploads to the Catalyst
+ *
+ * @param context - The request context containing form data and components
+ * @returns Upload result or error response
+ */
+export async function entitiesHandler(
+  context: FormHandlerContextWithPath<'logs' | 'metrics' | 'authorizations' | 'linker', '/content/entities'>
+): Promise<IHttpServerComponent.IResponse> {
+  const {
+    formData,
+    components: { logs, metrics, authorizations, linker }
+  } = context
+
+  const logger = logs.getLogger('entities-handler')
+
+  try {
+    logger.info('POST: /content/entities')
+
+    // Parse the auth chain from the form data
+    const authChainField = formData.fields.authChain
+    if (!authChainField) {
+      throw new ForbiddenError('No auth chain provided')
+    }
+
+    const authChain = JSON.parse(authChainField.value)
+
+    // Validate the signature
+    const validationResult = await linker.validateAuthChain(authChain)
+    if (!validationResult.ok) {
+      throw new ForbiddenError('Invalid auth chain')
+    }
+
+    // Check authorization
+    const signerAddress = validationResult.signerAddress
+    const authorizationResult = await authorizations.checkAuthorization(signerAddress)
+    if (!authorizationResult.authorized) {
+      throw new ForbiddenError('Address not found')
+    }
+
+    logger.info('Signer found in authorizations list', { signerAddress })
+
+    // Get the entity ID
+    const entityIdField = formData.fields.entityId
+    if (!entityIdField) {
+      throw new InvalidRequestError('Missing entityId')
+    }
+
+    const entityId = entityIdField.value
+
+    // Get the entity file
+    const entityFile = formData.files[entityId]
+    if (!entityFile) {
+      throw new InvalidRequestError('Missing entity file')
+    }
+
+    const entityContent = entityFile.value.toString('utf-8')
+    const entity = JSON.parse(entityContent)
+
+    // Check parcel access
+    const accessResult = await authorizations.checkParcelAccess(signerAddress, entity.pointers)
+    if (!accessResult.hasAccess) {
+      throw new ForbiddenError(
+        `Missing access for ${accessResult.missingParcels.length} parcels:\n${accessResult.missingParcels.join('; ')}`
+      )
+    }
+
+    // Upload to Catalyst
+    const uploadResult = await linker.uploadToCatalyst(entityId, formData.files)
+
+    if (!uploadResult.success) {
+      throw new Error(uploadResult.error ?? 'Upload failed')
+    }
+
+    metrics.increment('linker_entity_upload_counter', { status: 'success' })
+
+    return {
+      status: 200,
+      body: uploadResult.response as unknown as IHttpServerComponent.IResponse['body']
+    }
```

Comment:
Two questions about the new response:
1. Are we ok with breaking the response interface? Before, we were responding with plain text.
2. The previous handler was setting this header `res.setHeader('X-Extend-CF-Timeout', 10)`, should we include it here too?

Reply:
The response is returning exactly the same response as what the Catalyst would, which was the original behavior. The thing is that we don't really know what it's going to be responding, so that's why I have to re-cast it as body.

The previous handler set the `X-Extend-CF-Timeout` as a response header, but according to what I've been reading, it wasn't useful. I couldn't really find any official documentation about this header, but the AI says that it was to help Cloudflare know to time out later in time when a request was done from the server to another server.